samples/snippets/csharp/VS_Snippets_CLR_System/system.collections.generic.list.sort/cs/program.cs

```diff
@@ -1,129 +1,111 @@
 ﻿//<snippet1>
 using System;
 using System.Collections.Generic;
-// Simple business object. A PartId is used to identify the type of part
-// but the part name can change.
-public class Part : IEquatable<Part> , IComparable<Part>
+
+// Simple business object. A PartId is used to identify the
+// type of part but the part name can change.
+public class Part : IEquatable<Part>, IComparable<Part>
 {
     public string PartName { get; set; }
 
     public int PartId { get; set; }
 
-    public override string ToString()
-    {
-        return "ID: " + PartId + "   Name: " + PartName;
-    }
-    public override bool Equals(object obj)
-    {
-        if (obj == null) return false;
-        Part objAsPart = obj as Part;
-        if (objAsPart == null) return false;
-        else return Equals(objAsPart);
-    }
-    public int SortByNameAscending(string name1, string name2)
-    {
+    public override string ToString() => 
+        $"ID: {PartId}   Name: {PartName}";
 
-        return name1.CompareTo(name2);
-    }
+    public override bool Equals(object obj) => 
+        obj is null
+            ? false
+            : !(obj is Part part)
+                ? false
+                : Equals(part);
```

Comment:
This can be simplified because `obj is Part part` is `false` if `obj` is `null`:

```suggestion
        (obj is Part part)
                ? Equals(part)
                : false;
```